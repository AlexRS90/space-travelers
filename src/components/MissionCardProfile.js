import React from 'react';
import PropTypes from 'prop-types';

const MissionCardProfile = ({ reserMisson }) => (
  <tbody>
    <tr>
      <td>{reserMisson.mission_name}</td>
    </tr>
  </tbody>
);

MissionCardProfile.propTypes = {
  reserMisson: PropTypes.node.isRequired,
};

export default MissionCardProfile;
