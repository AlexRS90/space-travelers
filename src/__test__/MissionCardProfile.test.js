import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import MissionCardProfile from '../components/MissionCardProfile';

afterEach(() => {
  cleanup();
});

const missions = [
  {
    mission_name: 'Thaicom',
    mission_id: 'EFDRT',
    reserved: true,
  },
  {
    mission_name: 'Thaicom',
    mission_id: 'HJUIY',
    reserved: false,
  },
  {
    mission_name: 'Falcon',
    mission_id: 'EFWERT',
    reserved: true,
  },
];
const reserMisson = missions.filter((mission) => mission.reserved === true);

describe('render component to the screen', async () => {
  test('render mission card to the screen', async () => {
    render(<MissionCardProfile reserMisson={reserMisson} />);
    const missionCardElement = await screen.getByTestId('profile-card');
    expect(missionCardElement).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<MissionCardProfile reserMisson={reserMisson} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
