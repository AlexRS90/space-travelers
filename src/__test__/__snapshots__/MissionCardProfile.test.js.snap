// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component to the screen matches snapshot 1`] = `
<tbody
  data-testid="profile-card"
>
  <tr>
    <td />
  </tr>
</tbody>
`;
